src/app/components/common/address_search/addresssearch.tsx

```diff
@@ -0,0 +1,80 @@
+/** @jsxImportSource @emotion/react */
+import React, { useState } from "react";
+import axios from "axios";
+import "./addresssearch.css"
+
+const AddressSearch = () => {
+    const [text, setText] = useState("");
+    const [address, setAddress] = useState({
+        address1:"",
+        address2:"",
+        address3:"",
+        kana1:"",
+        kana2:"",
+        kana3:"",
+        prefcode:"",
+        zipcode:""
+    });
+    const [isLoading, setIsLoading] = useState(false);
+
+    const onClickFetchData = async () => {
+        setIsLoading(true);
+        await axios
+          .get("https://zipcloud.ibsnet.co.jp/api/search", {
+            params: {
+              zipcode: text,
+            },
+          })
+          .then((res) => {
+            setAddress({
+                address1: res.data.results[0].address1,
```

Comment:
To specify magic number is not good.
If you can't understand what I said, please inform me.